rmw_dds_common/src/qos.cpp

```diff
@@ -0,0 +1,313 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rmw_dds_common/qos.hpp"
+
+#include <cstring>
+
+#include "rcutils/snprintf.h"
+#include "rmw/error_handling.h"
+#include "rmw/qos_profiles.h"
+
+namespace rmw_dds_common
+{
+
+static bool
```

Comment:
IMO, these rmw_time-related operators could live in a publicly-available header as potentially useful functionality outside of this QoS comparison functionality.

Reply:
I agree. But, there is some on-going discussion about deprecating `rmw_time_t` (https://github.com/ros2/rmw/pull/298) that would require changes here. I'd rather wait to see how that plays out before abstracting these operators to a common place.

Reply:
Ok, sounds good.